### PR TITLE
Sync avatar

### LINKED ISSUE #31: [Syncing] Avatar
## Points to note
- Component in action - `Avatar.js`
- Github Urls to this component:
  * [neeto-ui/Avatar.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/Avatar.js)
  * [NitroUI/Avatar.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Avatar.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `Avatar.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**

## Diff
```diff
--- NitroUI/Avatar.js	2020-11-28 16:54:23.805687700 +0530
+++ neetoui-official/Avatar.js	2020-11-28 16:49:18.734801800 +0530
@@ -1,9 +1,14 @@
 import React from "react";
 import classnames from "classnames";
-import { getInitials } from "common";
+const getInitials = name => {
+  var initials = name.match(/\b\w/g) || [];
+  initials = ((initials.shift() || "") + (initials.pop() || "")).toUpperCase();
+  return initials;
+};
 
 const Avatar = ({ contact, className, size, isOnline, activity }) => {
   const tailwindColors = [
+    "gray",
     "red",
     "orange",
     "green",
@@ -19,9 +24,7 @@
     width: size,
     height: size,
     minHeight: size,
-    minWidth: size,
-    fontSize: Math.ceil(size * 0.375),
-    lineHeight: `${size}px`
+    minWidth: size
   };
 
   const activityIcons = {
@@ -46,17 +49,17 @@
   const renderUserInitial = userName => {
     const userInitial = getInitials(userName);
     const charCode =
-      (userInitial[0] ? userInitial.charCodeAt(0) : 0) +
-      (userInitial[1] ? userInitial.charCodeAt(1) : 0);
+      (userInitial ? userInitial.charCodeAt(0) : 0) +
+      (userInitial ? userInitial.charCodeAt(1) : 0);
     const bgColor =
       tailwindColors[(charCode % 65) % tailwindColors.length] || "gray";
-    const backgroundClass = `bg-${bgColor}-300`;
+    const backgroundClass = `bg-${bgColor}-400`;
 
     return (
       <div
         className={classnames([
           "relative flex flex-row items-center justify-center",
-          "text-white font-medium rounded-full",
+          "text-sm text-white font-medium rounded-full",
           backgroundClass,
           className
         ])}
@@ -71,7 +74,7 @@
 
   const renderUserImage = profileImagePath => {
     return (
-      <div className={classnames(["relative", className])} style={style}>
+      <div className={className} style={style}>
         <img
           src={profileImagePath}
           className="object-cover w-full h-full overflow-hidden rounded-full"
@@ -81,11 +84,21 @@
     );
   };
 
-  const userName = contact.name ? contact.name : contact.profile_name;
+  const renderDefaultAvatar = () => 
+    <div className={className} style={style}>
+      <svg className="object-cover w-full h-full p-2 overflow-hidden text-white bg-purple-400 rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24"><path fill="none" d="M0 0h24v24H0z"/><path fill="#ffffff" d="M4 22a8 8 0 1 1 16 0H4zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6z"/></svg>
+      {isOnline && <Indicator />}
+    </div>
+
+
+  if(!!!contact){
+    return renderDefaultAvatar()
+  }
 
   if (contact.profile_image_path) {
     return renderUserImage(contact.profile_image_path);
   } else {
+    const userName = contact.name ? contact.name : contact.profile_name;
     return renderUserInitial(userName);
   }
 };
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #31 

Changes/Enhancements:
- Didn't copy paste NitroUI component as is, because I felt neeto-ui has done stuff better, like rendering default avatar etc
- Fixed issue with Indicator not being displayed over avatar when using default avatar. The following was the issue:
avatar-component-indicator-top-right-corner-render-default
- Handled cases where no contact info is provided/contact name is empty
- Handled case where activity attr is invalid
- Removed unnecessary conditions like `userInitial ? `

@edwinbbu _a
Please review the PR and merge if everything looks good. Thanks

### CODE REVIEW

**Comment by goutham-subramanyam** [comment]
@yedhink _a

1. We need to include `Activity` in `renderUserImage` as well.
2. The `isOnline` prop is being updated to `status` ([Ref](https://github.com/bigbinary/neeto-help-web/pull/8592/files#diff-53ad57a285a90f0d78b1b57db2302fa4e41c22b830fa333906c88531bc8c749e)). We need to update that as well.

Also, please tag in me all the syncing PRs. Thanks.

**Comment by yedhink** [comment]
@vinay0x _a

Please review the PR.

How exactly should we deal with conflicts here? The latest bundles are being in conflict. Manually fixing the conflicts in bundles is a task since the file content is too big to diff. Can we merge this overwrite the bundles in master?

cc: @goutham-subramanyam

**Comment by vinay0x** [comment]
@yedhink _a Rebase with master and ignore the conflicts in bundles. You can accept the master version But make sure that the lib folder contents are latest in your branch. Then create a build and push it.

**Comment by yedhink** [comment]
@vinay0x _a

Have rebased.

Let me know if this is good to be merged.

File: lib/components/Avatar.js [hunk 1]
```diff
@@ -49,49 +63,56 @@ const Avatar = ({ contact, className, size, isOnline, activity }) => {
...
-          "text-sm text-white font-medium rounded-full",
+          "text-white font-medium rounded-full",
           backgroundClass,
           className
         ])}
         style={style}
       >
         {userInitial}
-        {isOnline && <Indicator />}
+        {status && <Indicator />}
```
Comment (goutham-subramanyam) [inline lib/components/Avatar.js:83]: The `Indicator` needs to be updated like in that PR.

**Review by goutham-subramanyam — CHANGES_REQUESTED** [review]
@yedhink

**Comment by yedhink** [comment]
@goutham-subramanyam _a

I have made the changes. Please review and merge if everything looks good.

File: lib/components/Avatar.js [hunk 2]
```diff
@@ -49,49 +70,56 @@ const Avatar = ({ contact, className, size, isOnline, activity }) => {
...
-          "text-sm text-white font-medium rounded-full",
+          "text-white font-medium rounded-full",
           backgroundClass,
           className
         ])}
         style={style}
       >
         {userInitial}
-        {isOnline && <Indicator />}
+        <Indicator status={status} />
```
Comment (goutham-subramanyam) [inline lib/components/Avatar.js:90]: We should have the `status` check. Else an unnecessary `<span>` will be rendered all the time right.

File: lib/components/Avatar.js [hunk 3]
```diff
@@ -49,49 +70,56 @@ const Avatar = ({ contact, className, size, isOnline, activity }) => {
...
 
-  const renderDefaultAvatar = () => 
-    <div className={className} style={style}>
-      <svg className="object-cover w-full h-full p-2 overflow-hidden text-white bg-purple-400 rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24"><path fill="none" d="M0 0h24v24H0z"/><path fill="#ffffff" d="M4 22a8 8 0 1 1 16 0H4zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6z"/></svg>
-      {isOnline && <Indicator />}
-    </div>
+  const renderDefaultAvatar = () => {
+    return (
+      <div className={classnames(["relative", className])} style={style}>
+        <svg className="object-cover w-full h-full p-2 overflow-hidden text-white bg-purple-400 rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
```
Comment (goutham-subramanyam) [inline lib/components/Avatar.js:112]: If we have added remixicons, please replace with that. Else its fine.

**Review by goutham-subramanyam — CHANGES_REQUESTED** [review]
@yedhink _a

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Please review and merge if everything looks good.

The status check is done within `Indicator` component.

**Review by goutham-subramanyam — APPROVED** [review]

**Pull request merged**